### ISSUE TITLE
reject HostKey with some servers

### ISSUE THREAD

**Issue by artjomsk** [mwiede/jsch#13]
Hello!

The issue is starting from 0.1.58

Old hostkeys (ssh-rsa) are not working. Getting the following error in log:
```
JSCHLogger] (default task-5) kex: server: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1
JSCHLogger] (default task-5) kex: server: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
JSCHLogger] (default task-5) kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc,blowfish-cbc,cast128-cbc,3des-cbc
JSCHLogger] (default task-5) kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc,blowfish-cbc,cast128-cbc,3des-cbc
JSCHLogger] (default task-5) kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
JSCHLogger] (default task-5) kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
JSCHLogger] (default task-5) kex: server: none,zlib@openssh.com
JSCHLogger] (default task-5) kex: server: none,zlib@openssh.com
JSCHLogger] (default task-5) kex: server: 
JSCHLogger] (default task-5) kex: server: 
JSCHLogger] (default task-5) kex: client: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
JSCHLogger] (default task-5) kex: client: ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,rsa-sha2-512,rsa-sha2-256,rsa-sha2-512,ssh-rsa
JSCHLogger] (default task-5) kex: client: aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
JSCHLogger] (default task-5) kex: client: aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
JSCHLogger] (default task-5) kex: client: hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
JSCHLogger] (default task-5) kex: client: hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
JSCHLogger] (default task-5) kex: client: none
JSCHLogger] (default task-5) kex: client: none
JSCHLogger] (default task-5) kex: client: 
JSCHLogger] (default task-5) kex: client: 
JSCHLogger] (default task-5) kex: algorithm: ecdh-sha2-nistp256
JSCHLogger] (default task-5) kex: host key algorithm: ecdsa-sha2-nistp256
JSCHLogger] (default task-5) kex: server->client cipher: aes128-ctr MAC: hmac-sha2-256-etm@openssh.com compression: none
JSCHLogger] (default task-5) kex: client->server cipher: aes128-ctr MAC: hmac-sha2-256-etm@openssh.com compression: none
JSCHLogger] (default task-5) SSH_MSG_KEX_ECDH_INIT sent
JSCHLogger] (default task-5) expecting SSH_MSG_KEX_ECDH_REPLY
JSCHLogger] (default task-5) Disconnecting from [xxx] port 22
```

Using host key with type **ecdsa-sha2-nistp256** is working. As well, some other sftp servers are working with old host keys - on log I see `kex: host key algorithm: rsa-sha2-512`

Is it possible to support old host keys? Tried changing different config options with no luck.

**UPD**: server version which is failing is OpenSSH_7.4

**Comment by artjomsk** [comment]
OK, figured out that it can be configured with **server_host_key** option

**Comment by norrisjeremy** [comment]
Hi @artjomsk, I am interested to better understand the failures you were seeing.
If I understand correctly, the problem was occurring with a server running OpenSSH 7.4? Do you know what Linux distribution it was running?
Also, would you be able to provide a copy of the sshd_config file from the server?

**Comment by norrisjeremy** [comment]
Ok, I believe that I have reproduced this issue locally.
@mwiede This issue is due to the change of default `PubkeyAcceptedKeyTypes` I submitted here: [04d3ff1](https://github.com/mwiede/jsch/commit/04d3ff11187719e15ee91a73d104fe11f5026ef4#diff-a7096f419f397fd75a0bfc29ad53cb793e3245bb394b6beeb20f9eb7c485abcbR144).
I will submit a PR later today to try and rectify this problem. I should have been more careful with changing the `PubkeyAcceptedKeyTypes` values in order to maintain backwards compatibility.
